enum.c

```diff
@@ -2824,30 +2840,23 @@ each_val_i(RB_BLOCK_CALL_FUNC_ARGLIST(i, p))
 
 /*
  *  call-seq:
- *     enum.each_entry { |obj| block }  -> enum
- *     enum.each_entry                  -> an_enumerator
+ *    each_entry(*args) {|element| ... } -> self
+ *    each_entry(*args)                  -> enumerator
  *
- *  Calls <i>block</i> once for each element in +self+, passing that
- *  element as a parameter, converting multiple values from yield to an
- *  array.
+ *  Calls the given block with each element,
+ *  converting multiple values from yield to an array; returns +self+:
  *
- *  If no block is given, an enumerator is returned instead.
- *
- *     class Foo
- *       include Enumerable
- *       def each
- *         yield 1
- *         yield 1, 2
- *         yield
- *       end
- *     end
- *     Foo.new.each_entry{ |o| p o }
+ *    a = []
```

Comment:
I think the sense of the method not only in description, but in removed example (_converted that example to new style_)

```ruby
a = []
class Foo
  include Enumerable
  def each
    yield 1
    yield 1, 2
    yield
  end
 end
 Foo.new.each_entry{ |o| a.push(o) } 
# => self (produced by Foo.new)
a # => [1, [1, 2], nil]
```
Briefly: example must contain yield with multiple attributes

Reply:
```ruby
a = []
def calls_various_blocks
  yield 1
  yield 1, 2
  yield
end
to_enum(:calls_various_blocks).each_entry{ |e| a.push(e)}
# => #<Enumerator: ...>
a #=> [1, [1, 2], nil]
```